### PR TITLE
Explore: Move back / forward with browser buttons now works

### LINKED ISSUE #14301: Explore: cannot use backward/forward (browser) actions
Doing changes in the explore mode like changing query or changing data source updates the URL. 

Going backward via browser updates the URL but the UI stay the same.

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR makes it possible to navigate backward and forward in browser using Explore.

**Which issue(s) this PR fixes**:
Fixes #14301

**Special notes for your reviewer**:
Checkout out the branch and try it out

### CODE REVIEW

**Review by davkal — APPROVED** [review]
LGTM

Works really well, and this is a well-written PR. Outstanding!

Small nit: Not happy with `refresh...`, as it's overlapping with browser refreshs. How about "save", or "persist"?

Found unrelated bug which will surface more when buttons are used: #16154

**Comment by hugohaggmark** [comment]
Just found out that back/forward doesn't work well with Graphite datasource #15862
What about `update` instead of `refresh`?

**Comment by torkelo** [comment]
+1 for update

Great PR

**Review by torkelo — APPROVED** [review]

File: public/test/core/thunk/thunkTester.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  whenThunkIsDispatched: (...args: any) => ThunkThen;
+}
+
+export interface ThunkThen {
+  thenDispatchedActionsEqual: (actions: Array<ActionOf<any>>) => ThunkWhen;
+  thenDispatchedActionsAreEqual: (callback: (actions: Array<ActionOf<any>>) => boolean) => ThunkWhen;
+  thenThereAreNoDispatchedActions: () => ThunkWhen;
+}
+
+export const thunkTester = (initialState: any): ThunkGiven => {
```
Comment (dprokop) [inline public/test/core/thunk/thunkTester.ts]: Just wanted to say that using this testing API's you propose for redux is so much fun and easy!

**Review by dprokop — APPROVED** [review]

**Pull request merged**